File: Java/Negative split.java
import java.util.List;
import java.util.Scanner;
import java.util.ArrayList;

public class Main
{
    private static int timeToSeconds(String time) 
    {
        String[] parts = time.split(":");
        return Integer.parseInt(parts[0]) * 3600 + Integer.parseInt(parts[1]) * 60 + Integer.parseInt(parts[2]);
    }

    public static void analyzeResults(int n, List<String> results) 
    {
        int a = 0, b = 0, c = 0, d = 0;
        for (String result : results) 
        {
            String[] timesStr = result.split(" ");
            int[] times = new int[6];
            for (int i = 0; i < 6; ++i) 
            {
                times[i] = timeToSeconds(timesStr[i]);
            }
            if (times[0] > (times[3] - times[2]) && times[1] >= times[0] && times[2] >= times[1] && times[3] >= times[2]) 
            {
                ++b;
                if (times[4] <= times[5]) 
                    ++a;
            } 
            else 
            {
                ++d;
                if (times[4] <= times[5]) 
                    ++c;
            }
        }
        System.out.print(a + "/" + b + " " + c + "/" + d);
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int n = Integer.parseInt(input.nextLine());
        List<String> results = new ArrayList<>();
        for (int i = 0; i < n; ++i)
            results.add(input.nextLine());
        analyzeResults(n, results);
        input.close();
    }
}

File: Python/Negative split.py
def parse_time(time_str):
    h, m, s = map(int, time_str.split(':'))
    return h * 3600 + m * 60 + s

def analyze_results(results):
    b = a = d = c = 0
    for result in results:
        times = list(map(parse_time, result.split()))
        if all(times[i] - times[i-1] <= times[i-1] - times[i-2] for i in range(2, 5)) and times[0] > times[3] - times[2]:
            b += 1
            if times[4] <= times[5]:
                a += 1
        else:
            d += 1
            if times[4] <= times[5]:
                c += 1
    return f"{a}/{b} {c}/{d}"

n = int(input())
results = [input() for _ in range(n)]
print(analyze_results(results))